editor/include/ImGuiCore/ImGuiManager.hxx

```diff
@@ -14,10 +14,10 @@ class ImGuiManager
 {
 
 public:
-    ImGuiManager(const pivot::ecs::SceneManager &sceneManager, pivot::Engine &engine)
+    ImGuiManager(const pivot::ecs::SceneManager &sceneManager, pivot::Engine &engine, const std::filesystem::path &asset_dir)
         : m_sceneManager(sceneManager), m_engine(engine)
     {
-        for (const auto &directoryEntry: std::filesystem::recursive_directory_iterator(PIVOT_EDITOR_ICON_PATH)) {
+        for (const auto &directoryEntry: std::filesystem::recursive_directory_iterator(asset_dir.string() + "/Editor")) {
```

Comment:
You can do this instead of converting it back and forth between string to path
```suggestion
        for (const auto &directoryEntry: std::filesystem::recursive_directory_iterator(asset_dir / "Editor")) {
```